### ISSUE TITLE
[Bugs] Retry Policy isn't properly handled 

### ISSUE THREAD

**Issue by benriou** [go-to-k/cls3#178]
**Describe the bug**

The ReadMe tells about a retry mechanism while getting slowdown errors from S3 API. 

However, CLS3 stops operations as shown below. 

```
[root@ip-172-31-89-44 cls3]# ./cls3 --bucketName OBFUSCATED
DBG Debug mode...
INF OBFUSCATED Checking...
Clearing... 145000 objects
ERR [resource OBFUSCATED] operation error S3: ListObjectVersions, exceeded maximum number of attempts, 3, https response error StatusCode: 503, RequestID: HW54VYVJQ8HA1N31, HostID: L2I//VtHpCedDT+8tKW3bjVfjq3vhL9J1dKBLVrcUK3osx81AxRV2TU8WJdjxOwDwsmMoMFaDFZAFaMXrqKkYgZR56naoauwS9ySm5HWqyw=, api error SlowDown: Please reduce your request rate.
[root@ip-172-31-89-44 cls3]#
```

**To Reproduce**

Clean a bucket from an EC2 using the latest CLS3 (manually build from source code : master).

**Current workaround**

**Expected behavior**
The CLS3 program should try to restart. 

**Environment**
- OS: Amazon Linux 2023
- EC2 deployed on the VPC

**Comment by go-to-k** [comment]
Hi, @benriou 
Thank you for this report!

In fact, I had included a retry process in the DeleteObjects API but not in the ListObjectVersions API. The information that it is also needed for that API was very helpful.

I will fix it this week. I would put in the same process as the DeleteObjects API.
Thanks.

**Comment by go-to-k** [comment]
@benriou 

I have fixed this issue and released v0.18.0 which includes it! Please try it out.

https://github.com/go-to-k/cls3/releases/tag/v0.18.0
